api/envoy/extensions/filters/http/oauth2/v3/oauth.proto

```diff
@@ -0,0 +1,77 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.oauth2.v3;
+
+import "envoy/extensions/transport_sockets/tls/v3/secret.proto";
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.oauth2.v3";
+option java_outer_classname = "OauthProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: OAuth]
+// OAuth :ref:`configuration overview <config_http_filters_oauth>`.
+// [#extension: envoy.filters.http.oauth2]
+//
+
+message OAuth2Credentials {
+  // The client_id to be used in the authorize calls. This value will be URL encoded when sent to the OAuth server.
+  string client_id = 1 [(validate.rules).string = {min_bytes: 1}];
+
+  // The secret used to retrieve the access token. This value will be URL encoded when sent to the OAuth server.
+  transport_sockets.tls.v3.SdsSecretConfig token_secret = 2
+      [(validate.rules).message = {required: true}];
+
+  // The secret used to perform the HMAC validation.
+  transport_sockets.tls.v3.SdsSecretConfig hmac_secret = 3
+      [(validate.rules).message = {required: true}];
```

Comment:
I don't think HMAC is a required component for OAuth 2.0 spec?

Reply:
@rgs @jparise can either of you speak to this? I'd be curious to know what the behavior of the system would be without the HMAC component

Reply:
@rgs1 😉

Reply:
HMAC isn't required by the [OAuth 2.0 spec](https://tools.ietf.org/html/rfc6749) because token/secret construction is left to the implementation. I think it's fine for us to require HMAC here if that's how the tokens/secrets are formed.

Reply:
@jparise It is how tokens/secrets are formed now, though how this could be evolved to support other implementation, for example OIDC with ID token signature, in the future?

Reply:
Yes, that's a valid point. This could be future-proofed schematically by making this optional or wrapping it a `oneof` of messages types that describe the token formation method.

Reply:
I wrapped this in a oneof, lmk if that seems like a good direction